server/container/guice/data-ldap/src/main/java/org/apache/james/data/LdapUsersRepositoryModule.java

```diff
@@ -43,6 +45,7 @@ public void configure() {
         bind(UsersRepository.class).to(ReadOnlyUsersLDAPRepository.class);
         bind(DelegationStore.class).to(NaiveDelegationStore.class);
         bind(Authorizator.class).to(UserRepositoryAuthorizator.class);
+        bind(HealthCheck.class).to(LdapHealthCheck.class);
```

Comment:
No this needs to be done through the multi-binder:

```suggestion
        Multibinder.newSetBinder(binder(), HealthCheck.class).addBinding().to(LdapHealthCheck.class);
```

Reply:
Ah right! because we need to inject the list of health check implementations somewhere else in the code.